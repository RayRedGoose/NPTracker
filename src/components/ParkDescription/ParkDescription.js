import './ParkDescription.scss'
import React from 'react'
import arrow from 'assets/more.svg'

const ParkDescription = (props) => {
  const { image, fullName, url, states, description } = props

  return (
    <section className="park-info">
      <div style={{backgroundImage: `url(${image})`}}></div>
      <header>
        <h2>{fullName}</h2>
        <a href={url}><img src={arrow} alt="arrow"/></a>
      </header>
      <p className="location">{states}</p>
      <p className="desc">{description}</p>
    </section>
  )
}

export default ParkDescription
