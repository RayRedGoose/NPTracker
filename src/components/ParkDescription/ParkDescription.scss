@import "../../variables.scss";
@import "../../index.scss";

.park-info {
  grid-area: desc;
  display: grid;
  grid-template-areas: 'img head' 'img place' 'img desc';
  padding: 20px;
  height: 100%;
  &>div {
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
    grid-area: img;
    width: 20vw; height: 100%;
    margin-right: 20px;
  }
  header {
    grid-area: head;
    display: flex;
    justify-content: space-between;
    align-content: center;
    margin-bottom: 10px;
    h2 {
      color: $accent-color-primary-light;
      font-family: Salsa;
      letter-spacing: 0.05em;
      margin-right: 20px;
    }
  }
  .location {
    grid-area: place;
    @extend %location;
    margin-top: 0;
    font-size: 18px;
  }
  .desc {
    grid-area: desc;
    margin-top: 20px;
    font-size: 14px;
    line-height: 1.5em;
    height: 260px;
    overflow: scroll;
  }
}
