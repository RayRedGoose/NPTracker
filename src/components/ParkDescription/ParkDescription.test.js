import React from 'react'
import { shallow } from 'enzyme'
import ParkDescription from './ParkDescription'

describe('ParkDescription', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const widget = shallow(
      <ParkDescription
        image="https://some.url/image.jpg"
        fullName="Mount Rainier National Park"
        url="https://some.url"
        states="WA"
        description="Ascending to 14,410 feet above sea level, Mount Rainier stands as an icon in the Washington landscape." />
    )

    expect(widget).toMatchSnapshot()
  })
})
