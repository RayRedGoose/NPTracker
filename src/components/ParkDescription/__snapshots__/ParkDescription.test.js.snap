// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParkDescription should match the snapshot with all data passed in correctly 1`] = `
<section
  className="park-info"
>
  <div
    style={
      Object {
        "backgroundImage": "url(https://some.url/image.jpg)",
      }
    }
  />
  <header>
    <h2>
      Mount Rainier National Park
    </h2>
    <a
      href="https://some.url"
    >
      <img
        alt="arrow"
        src="more.svg"
      />
    </a>
  </header>
  <p
    className="location"
  >
    WA
  </p>
  <p
    className="desc"
  >
    Ascending to 14,410 feet above sea level, Mount Rainier stands as an icon in the Washington landscape.
  </p>
</section>
`;
